depends/packages/boost.mk

```diff
@@ -9,7 +9,7 @@ define $(package)_set_vars
 $(package)_config_opts_release=variant=release
 $(package)_config_opts_debug=variant=debug
 $(package)_config_opts=--layout=tagged --build-type=complete --user-config=user-config.jam
-$(package)_config_opts+=threading=multi link=static -sNO_BZIP2=1 -sNO_ZLIB=1
+$(package)_config_opts+=threading=multi link=static -sNO_COMPRESSION=1
```

Comment:
Even though this option shouldn't affect us, given we don't build iostreams, this is a nice simplification. Easier to pass `sNO_COMPRESSION` and not worry about new types being added in future, as we were already missing `NO_LZMA` and `NO_ZSTD`.